WooCommerce/src/main/kotlin/com/woocommerce/android/support/WooLogViewerActivity.kt

```diff
@@ -1,66 +1,32 @@
 package com.woocommerce.android.support
 
-import android.content.Context
 import android.content.Intent
 import android.os.Bundle
-import android.view.LayoutInflater
-import android.view.Menu
-import android.view.MenuItem
-import android.view.View
-import android.view.ViewGroup
-import android.widget.TextView
-import androidx.appcompat.app.AppCompatActivity
-import androidx.appcompat.widget.Toolbar
-import androidx.core.content.ContextCompat
-import androidx.recyclerview.widget.DividerItemDecoration
-import androidx.recyclerview.widget.LinearLayoutManager
-import androidx.recyclerview.widget.RecyclerView
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
 import com.woocommerce.android.R
-import com.woocommerce.android.R.layout
-import com.woocommerce.android.databinding.ActivityLogviewerBinding
-import com.woocommerce.android.extensions.setHtmlText
+import com.woocommerce.android.ui.compose.theme.WooThemeWithBackground
 import com.woocommerce.android.util.AppThemeUtils
 import com.woocommerce.android.util.WooLog
 import com.woocommerce.android.util.WooLog.T
 import com.woocommerce.android.util.copyToClipboard
-import com.woocommerce.android.widgets.AlignedDividerDecoration
 import org.wordpress.android.util.ToastUtils
-import java.lang.String.format
-import java.util.*
-
-class WooLogViewerActivity : AppCompatActivity() {
-    companion object {
-        private const val ID_SHARE = 1
-        private const val ID_COPY_TO_CLIPBOARD = 2
-    }
 
+class WooLogViewerActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        val binding = ActivityLogviewerBinding.inflate(layoutInflater)
-        setContentView(binding.root)
-
-        setSupportActionBar(binding.toolbar.toolbar as Toolbar)
-        supportActionBar?.setDisplayHomeAsUpEnabled(true)
-
-        setupRecyclerView(binding, setupRecyclerDivider())
-    }
-
-    private fun setupRecyclerDivider(): AlignedDividerDecoration {
-        val divider = AlignedDividerDecoration(
-            this,
-            DividerItemDecoration.VERTICAL, 0, clipToMargin = false
-        )
-        ContextCompat.getDrawable(this, R.drawable.list_divider)?.let { drawable ->
-            divider.setDrawable(drawable)
+        setContent {
+            WooThemeWithBackground {
+                WooLogViewerScreen(
+                    WooLog.logEntries,
+                    isDarkThemeActive = AppThemeUtils.isDarkThemeActive(this@WooLogViewerActivity),
+                    onBackPress = { onBackPressed() },
+                    onCopyButtonClick = { copyAppLogToClipboard() },
+                    onShareButtonClick = { shareAppLog() }
```

Comment:
Np no need to change. You can pass these as function references, instead as function calls wrapped in brackets.

```suggestion
                    onBackPress = ::onBackPressed,
                    onCopyButtonClick = ::copyAppLogToClipboard,
                    onShareButtonClick = ::shareAppLog
```